contrib/epee/src/readline_buffer.cpp

```diff
@@ -167,30 +165,42 @@ static int handle_enter(int x, int y)
   char* line = NULL;
```

Comment:
I'm not implying drawbacks. In general though, for this chunk of code, I'm not sure of the potential benefits outweighing soo many lines of extra code just to use a `unique_ptr`.

@vtnerd's initial comment was using `unique_ptr` "in case the string allocation throws". `rl_copy_text` internally uses c `malloc`, which never throws an exception, it just returns a null ptr if it cannot allocate.